protostar/commands/test/messages/broken_test_case_result_message.py

```diff
@@ -0,0 +1,74 @@
+from dataclasses import dataclass
+
+from protostar.io import StructuredMessage, LogColorProvider
+from protostar.testing import BrokenTestCaseResult
+
+from .utils import (
+    get_formatted_execution_time_human,
+    get_formatted_execution_time_structured,
+    get_formatted_file_path,
+    get_formatted_stdout,
+    get_formatted_metadata,
+)
+
+
+@dataclass
+class BrokenTestCaseResultMessage(StructuredMessage):
+    broken_test_case_result: BrokenTestCaseResult
+
+    def format_human(self, fmt: LogColorProvider) -> str:
+        result: list[str] = []
+        first_line_items: list[str] = [f"[{fmt.colorize('RED', 'BROKEN')}]"]
+
+        formatted_file_path = get_formatted_file_path(
+            file_path=self.broken_test_case_result.file_path, log_color_provider=fmt
+        )
+        first_line_items.append(
+            f"{formatted_file_path} {self.broken_test_case_result.test_case_name}"
+        )
+
+        info_items = [
+            get_formatted_execution_time_human(
+                execution_time=self.broken_test_case_result.execution_time,
+                log_color_provider=fmt,
+            )
+        ]
+
+        for key, value in self.broken_test_case_result.exception.execution_info.items():
+            info_items.append(f"{key}={fmt.bold(value)}")
+
+        if len(info_items) > 0:
+            info = ", ".join(info_items)
+            first_line_items.append(fmt.colorize("GRAY", f"({info})"))
+
+        result.append(" ".join(first_line_items))
+
+        result.append("\n")
+        result.append(str(self.broken_test_case_result.exception))
+        result.append("\n")
+
+        for metadata in self.broken_test_case_result.exception.metadata:
+            result.append(get_formatted_metadata(metadata))
+            result.append("\n")
+
+        result.extend(
+            get_formatted_stdout(
+                captured_stdout=self.broken_test_case_result.captured_stdout,
+                log_color_provider=fmt,
+            )
+        )
+
+        return "".join(result)
+
+    def format_dict(self) -> dict:
+        return {
+            "type": "test_case_result",
```

Comment:
I'm not sure this is a good idea. I think it's good to have a mutual type for all of these messages, it helps distinguish them from test collector results and test summaries.
I know we could have a mutual part of the type like `test_case_result_failed_test_case`, `test_case_result_skipped_test_case`, etc. But at the same time, we've got the `status` field which would basically duplicate the information on the specific type of message.

We could leave the type as is and make `status`es more specific, so instead of having `status: failed`, we'd have `status: failed_test_case`, etc. What do you think?

Reply:
>  think it's good to have a mutual type for all of these messages, it helps distinguish them from test collector results and test summaries.

This problem can be solved by common suffix.

For example:
`passed_test_case_result`
`failed_test_case_result`
`fuzz_failed_test_case_result`
`test_summary`

To find all test cases result you can run:
```py
[msg for msg in msgs if msg.type.ends_with("test_case_result")]
```

>We could leave the type as is and make statuses more specific, so instead of having status: failed, we'd have status: failed_test_case, etc. What do you think?

Protostar also returns fuzzing test results, and it might be helpful to have an easy way to distinguish such results from others. IMO `type` property is more fitting for storing that information.

Reply:
Yes, I understand, as I said:

> I know we could have a mutual part of the type like `test_case_result_failed_test_case`, `test_case_result_skipped_test_case`, etc.

So I know you can operate on the string to pull the info you're interested in.

I have yet another solution in mind. The case here is that you're thinking of the test type and I'm thinking about the message type. I think it'd be profitable to keep both pieces of information in the result. How about this then: We can keep the current type and rename it to the `message_type` or `result_type` and add another field, let's say `test_type` which would be `passed_test_case_result`/`failed_test_case_result`, etc. That way we could distinguish the results of test cases/test summary/test collector between each other and also distinguish the test case results between each other. I think that in this case, we could remove the `status` field. 
I think this solution is not expensive and it's going to be easier to use - I think that the possibility of separating different messages easily in the test output may be very handy.
What do you think?

Reply:
IMO, result types are a subset of message types, but "test_type" should be ok.

Reply:
[type => message_type, status => test_type](https://github.com/software-mansion/protostar/pull/1254/commits/e850cb777dc38a105486a4aaf1101905dc23e6b0)